### PR TITLE
Add support for Python 3.12 and enable it in CI

### LINKED ISSUE #24: Add development support for python 3.12
Thanks for wanting to report an issue you've found in valkey-py. Please delete this text and fill in the template below.  
It is of course not always possible to reduce your code to a small test case, but it's highly appreciated to have as much data as possible. Thank you!

**Version**: What valkey-py and what valkey version is the issue happening on?

**Platform**: Python 3.12 on MacOS (M-Series ARM devices)

**Description**: There are some package changes that need to be made to make development possible on python 3.12

- unpinning the version of invoke from 1.7.3 allowing version >=2.0.0 which removes usage of the `imp` module which was removed in Python 3.12
- replacing `distutils` which is being used to check for truthy-like strings using `strtobool` with a dedicated module still being maintained like `dirty-equals`

### PR DESCRIPTION
**Pull request by mkmkme**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [X] Do tests and lints pass with this change?
- [ ] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [ ] Is the new or changed code fully tested?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Is there an example added to the examples folder (if applicable)?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

This changeset finishes adding support for Python 3.12. This should close #24

### CODE REVIEW

**Comment by mkmkme** [comment]
Oh, I didn't know updating black would introduce _syntax errors_ for 3.8.

I'm going to drop this commit then. Not a biggie.

**Comment by ahmedsobeh** [comment]
> Oh, I didn't know updating black would introduce _syntax errors_ for 3.8.
> 
> I'm going to drop this commit then. Not a biggie.

yeah I faced this is another PR too. I think the line size limit on Black is different from whatever checks the linting in CI

**Comment by mkmkme** [comment]
> I think the line size limit on Black is different from whatever checks the linting in CI

For me it wasn't even about the line size. Apparently adding parenthesis to `with` statement and moving its internals to different lines wasn't valid in 3.8. Bizarre!

File: tasks.py [hunk 1]
```diff
@@ -25,56 +25,56 @@ def build_docs(c):
 
 
 @task
-def linters(c):
+def linters(c, color=False):
```
Comment (aiven-sal) [inline tasks.py:28]: Could you mention "fix #47" in the commit message for these changes?

Comment (mkmkme) [reply tasks.py:28]: Oh cool! Didn't know there was an issue for it. Added

**Review by aiven-sal — APPROVED** [review]

**Comment by kjaymiller** [comment]
So excited to see this merged!

**Pull request merged**